File: Java/Sorpresa_PorCasaHotel.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package civitas;

import java.util.ArrayList;

/**
 *
 * @author ilzk
 */
public class Sorpresa_PorCasaHotel extends Sorpresa {
    
    public Sorpresa_PorCasaHotel(){
                valor = 10;
    }
    
    @Override
    /**Aplica la sorpresa Por casa hotel al jugador actual
     * 
     */
    void aplicarAJugador(int actual, ArrayList<Jugador> todos) {
        if(jugadorCorrecto(actual, todos)) {
            informe(actual,todos);
            for (int i = 0; i < todos.get(actual).cantidadCasasHoteles(); i++) {
                todos.get(actual).modificarSaldo(valor);   
            }
        }
    }
    
    /**Sobreescribe el metodo toString
     * 
     */
    public String toString(){
        return "Por casa hotel";
    }
}

File: Ruby/sorpresa_por_casa_hotel.rb
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.

require_relative 'sorpresa'
module Civitas
  class Sorpresa_por_casa_hotel < Sorpresa
    def initialize
      @valor = 10
    end

    def aplicar_a_jugador(actual,todos)
      if (jugador_correcto(actual, todos))
        informe(actual, todos)
        for i in 1..todos[actual].cantidadCasasHoteles()
          todos[actual].modificarSaldo(@valor)
        end
      end
    end

    def to_s()
      "Por casa hotel"
    end

    public_class_method:new
  end
end